amardevsingh98/python-html-to-pdf
# Templates: Wisata Bill

### Description
1. Update PDF margins
2. Migrate wisatabill related utils to python
3. Create wisata bill template

### How to Test
1. Open chrome > chrome web inspector > network
2. Open https://project-exterior.com, login as backoffice
3. In dashboard, click `Backoffice` tab then choose `See Offline Bill`
4. In payment page, replace `on_behalf` query with one of these ID:
    | | |
    | --- | --- |
    | rodamas | 44 |
    | libing | 131 |
    | tumbakmas | 2117 |
    | sasainti | 64089 |
5. Download any billing statement
6. Network inspector should show request to `/api/print`
7. Copy value in `context` and paste into `data` in payload template below
    ```json5
    {
      "type": "wisata-bill",
      "format": "pdf",
      "filename": "My document",
      "header": {
        "logo_src": "https://cdn.wisata.app/f/52ee9453-c8c7-452d-9457-7674df571c8f.jpeg",
        "logo_alt": ""
      },
      "footer": {
        "show_download_app": false
      },
     "data": { /* ... */ }, // <-- put here
    }
    ```
8. Make request to `http://localhost:8000/print` with payload above
9. Compare billing statement downloaded from project-exterior and this project

### Test Result
| account | project-exterior | this project | notes |
| --- | --- | --- | --- |
| Rodamas July 2023 | [project-exterior.pdf](https://github.com/amardevsingh98/python-html-to-pdf/files/12330657/Wisata.Bill.Billing.Statement.-.Jul.2023.Wisata.App.22.pdf) | [response.pdf](https://github.com/amardevsingh98/python-html-to-pdf/files/12330658/response.pdf) | |
| Rodamas June 2023 | [project-exterior.pdf](https://github.com/amardevsingh98/python-html-to-pdf/files/12330662/Wisata.Bill.Billing.Statement.-.Jun.2023.Wisata.App.-.2023-08-14T063040.133.pdf) | [response.pdf](https://github.com/amardevsingh98/python-html-to-pdf/files/12330663/response.pdf) | has cancellation (both in diff period and in same period) |
